llvm/lib/Telemetry/Telemetry.cpp

```diff
@@ -0,0 +1,27 @@
+#include "llvm/Telemetry/Telemetry.h"
+
+namespace llvm {
+namespace telemetry {
+
+void TelemetryInfo::serialize(Serializer &serializer) const {
+  serializer.write("SessionId", SessionId);
+}
+
+Error Manager::dispatch(TelemetryInfo *Entry) {
+  if (Error Err = preDispatch(Entry))
+    return std::move(Err);
+
+  Error AllErrs = Error::success();
+  for (auto &Dest : Destinations) {
+    if (Error Err = Dest->receiveEntry(Entry))
+      AllErrs = joinErrors(std::move(AllErrs), std::move(Err));
```

Comment:
```suggestion
    AllErrs = joinErrors(std::move(AllErrs), Dest->receiveEntry(Entry));
```

(this should work because joinErrors is a no-op on success values)